docs/pages/developer-guide.md

```diff
@@ -936,6 +935,174 @@ In the example above, a server task can be accessed through the address:
+
+A certificate has following subject information:
+
+```
+CN=[pod-index-task].[service-name]
+O=Mesosphere, Inc
+L=San Francisco
+ST=CA
+C=US
+```
+
+Additional X.509 `Subject Alternative Names`, based on the pod `discovery` and `vip` confiagurations, are encoded into the certificate. If no `discovery` or port exposed over VIP is configured single the certificate comes with a single SAN.
+
+```
+DNSName([pod-index-task].[service-name].autoip.dcos.thisdcos.directory)
+```
+
+Each VIP exposed port will result in a new SAN entry:
+
+```
+DNSName([port-0-vip-prefix].[service-name].l4lb.thisdcos.directory)
+DNSName([port-1-vip-prefix].[service-name].l4lb.thisdcos.directory)
+```
+
+Providing a custom `prefix` under `discovery` level will result in replacing the default SAN `DNSName([pod-index-task].[service-name].autoip.dcos.thisdcos.directory)` with the configured name and pod index, i.e.:
+
+```
+DNSName([discovery-prefix-index].[service-name].autoip.dcos.thisdcos.directory)
+```
+
+This certificate configuration allows to run proper TLS hostname verification by client when task is accessed by one of [internal DNS names](https://docs.mesosphere.com/1.9/networking/dns-overview/).
+
+Each certificate is valid for **10 years**.
+
+### Artifacts format
+
+TLS artifacts can be provided to a task in two different formats:
+
+#### PEM
+
+Standard [PEM encoded](https://en.wikipedia.org/wiki/Privacy-enhanced_Electronic_Mail) certificate, private key and root CA certificate. Based on a file extension name it is possible to tell which artifact is in the file.
+
+A certificate file with **.crt** extension contains an end-entity certificate with possible chain of certificates leading to the root CA certificate. The root CA certificate is not part of end-entity certificate file.
+
+A **.key** file contains PEM encoded private key.
+
+A file with **.crt** extension contains root CA certificate without certificate chain.
+
+#### Java Keystore
+
+The [Java Keystore (JKS)](https://en.wikipedia.org/wiki/Keystore) is a repository of various certificates and private keys. When private key and certificate is requested a task will get 2 JKS format files.
+
+A **.keystore** is a JKS file that contains a private key with end-entity certificate and complete certificate chain including the root CA certificate. Certificate is stored under alias name **`default`**. The keystore and key are protected by single **`notsecure`** password.
```

Comment:
an end-entity certificate and a complete certificate chain, including the root CA certificate